### PR TITLE
Merge change 007

### LINKED ISSUE #10: Flash File Version Number ID
Requirement: Need way to identify code on PACMod component.
Message Name: FLASH_FILE_VERSION_XX
Message ID (Decimal): 0x1032
Message ID (Hexadecimal): 0x408
Frame Format: Standard
Data Length Code: 8
TX Node: PACMOD
~~Message Rate: 30 Hz~~
***Message Rate: 1 Hz***

|Name|Type|Bitpos|Length|Factor|Offset|Min|Max|Unit|
|---|---|---|---|---|---|---|---|---|
|MAJOR<sup>1</sup>|Unsigned|7|8|1|0|0|255||
|MINOR<sup>1</sup>|Unsigned|16|8|1|0|0|255||
|PATCH<sup>1</sup>|Unsigned|23|8|1|0|0|255||
|BRANCH<sup>2</sup>|Unsigned|31|8|1|0|0|255||
|CHANGE<sup>3</sup>|Unsigned|39|8|1|0|0|255||

1 Defined by Semantic Versioning 2.0.0.
2 Non-sequential number referencing a branch in version control.
3 Sequential number referencing a version on the branch.

|Name|Type|Bitpos|Length|Factor|Offset|Min|Max|Unit|
|---|---|---|---|---|---|---|---|---|
|MAJOR<sup>1</sup>|Unsigned|7|8|1|0|0|255||
|MINOR<sup>1</sup>|Unsigned|15|8|1|0|0|255||
|PATCH<sup>1</sup>|Unsigned|23|8|1|0|0|255||
|CHAR0|Unsigned|31|8|1|0|0|255||
|CHAR1|Unsigned|39|8|1|0|0|255||
|CHAR2|Unsigned|47|8|1|0|0|255||
|CHAR3|Unsigned|55|8|1|0|0|255||
|CHAR4|Unsigned|63|8|1|0|0|255||

### LINKED ISSUE #11: Missing Value Lists on _FAULT Signals
Requirement: Add value definitions for all _RPT messages on all _FAULT signals.
Message Name: All _RPT messages
Message ID (Decimal): N/A
Message ID (Hexadecimal): N/A
Frame Format: Standard
Data Length Code: 8
TX Node: PACMOD
Message Rate: 30 Hz

|Name|Type|Bitpos|Length|Factor|Offset|Min|Max|Unit|
|---|---|---|---|---|---|---|---|---|
|ENABLED | Unsigned | 0 | 1 | 1 | 0 | 0 | 1 |  
|OVERRIDE_ACTIVE | Unsigned | 1 | 1 | 1 | 0 | 0 | 1 |  
|COMMAND_OUTPUT_FAULT | Unsigned | 2 | 1 | 1 | 0 | 0 | 1 |  
|INPUT_OUTPUT_FAULT | Unsigned | 3 | 1 | 1 | 0 | 0 | 1 |  
|OUTPUT_REPORTED_FAULT | Unsigned | 4 | 1 | 1 | 0 | 0 | 1 |  
|PACMOD_FAULT | Unsigned | 5 | 1 | 1 | 0 | 0 | 1 |  
|VEHICLE_FAULT | Unsigned | 6 | 1 | 1 | 0 | 0 | 1 |  
|COMMAND_TIMEOUT 1 | Unsigned | 7 | 1 | 1 | 0 | 0 | 1 |  
|MANUAL_INPUT | Unsigned | 15 | 16 | 0.001 | 0 | 0 | 1 |  
|COMMANDED_VALUE | Unsigned | 31 | 16 | 0.001 | 0 | 0 | 1 |  
|OUTPUT_VALUE | Unsigned | 47 | 16 | 0.001 | 0 | 0 | 1

1 0=NO_FAULT, 1=FAULT

The current layout is shown above. This has a value list for COMMAND_TIMEOUT. The values for COMMAND_TIMEOUT should be 0=NO_TIMEOUT and 1=TIMEOUT whereas the _FAULT signals have no value lists which should be the value list shown above.

### LINKED ISSUE #21: VIN, Firmware, Watchdog Report
# Original Request Modified!! See Below !!

## VIN Report
Requirement: Add Transmission Rate info to DBC for VIN Report
Message Name: VIN Report
Message ID (Decimal): 1044
Message ID (Hexadecimal): 0x414
Frame Format: Standard
Data Length Code: 8
TX Node: PACMOD
Message Rate: 1 Hz (Proposed)

## Firmware Version Report
Requirement: Add Transmission Rate info to DBC for Firmware Version Report
Message Name: Firmware Version Report
Message ID (Decimal):
Message ID (Hexadecimal):
Frame Format: Standard
Data Length Code: 8
TX Node: PACMOD
Message Rate: 1 Hz (Proposed)

## Watchdog Report
Requirement: Add Transmission Rate and DBC info for Watchdog Report
Message Name: Watchdog Report
Message ID (Decimal): 1536 (Proposed)
Message ID (Hexadecimal): 0x600 (Proposed)
Frame Format: Standard
Data Length Code: 8
TX Node: PACMOD
Message Rate: 10 Hz (Proposed)

### PR DESCRIPTION
**Pull request by asaxsma**
This closes #10, closes #11, and closes #21.

### CODE REVIEW

**Review by mlemm99 — DISMISSED** [review]

File: as_pacmod.dbc [hunk 1]
```diff
@@ -636,15 +671,66 @@ CM_ SG_ 34 COUNTER "Counter shall have the value of 0 with the first message tra
...
+CM_ SG_ 1035 BUILD0 "BUILD0-BUILD3 shall represent the software build number and shall be constrained to characters A-Z and 0-9.";
+CM_ SG_ 1035 BUILD1 "BUILD0-BUILD3 shall represent the software build number and shall be constrained to characters A-Z and 0-9.";
+CM_ SG_ 1035 BUILD2 "BUILD0-BUILD3 shall represent the software build number and shall be constrained to characters A-Z and 0-9.";
+CM_ SG_ 1035 BUILD3 "BUILD0-BUILD3 shall represent the software build number and shall be constrained to characters A-Z and 0-9.";
 BA_DEF_ SG_  "GenSigCycleTime" INT 0 60000;
 BA_DEF_ BO_  "GenMsgCycleTime" INT 0 60000;
 BA_DEF_ SG_  "GenSigSendType" ENUM  "Cyclic ","OnWrite ","OnWriteWithRepetition ","OnChange ","OnChangeWithRepetition ","IfActive ","IfActiveWithRepetition ","NoSigSendType";
 BA_DEF_ BO_  "GenMsgSendType" ENUM  "Cyclic ","NotUsed ","NotUsed ","NotUsed ","NotUsed ","NotUsed ","NotUsed ","IfActive ","NoMsgSendType ";
+BA_DEF_ BO_  "Source_Address_Type" ENUM  "NO_SOURCE_ADDRESS","SOURCE_ADDRESS";
+BA_DEF_ BO_  "Source_Address" INT 0 255;
```
Comment (JWhitleyWork) [inline as_pacmod.dbc]: I'm not a fan of calling this SOURCE_ADDRESS. It makes an implication that we are using this for purposes other than as a component designation. Can we call it something like COMPONENT_ID or COMPONENT_ADDRESS?

Comment (asaxsma) [reply as_pacmod.dbc]: Sounds reasonable to me.

Comment (JWhitleyWork) [reply as_pacmod.dbc]: If you make this change and push it to this branch, it'll automatically update the PR.

**Comment by asaxsma** [comment]
Done.

**Review by JWhitleyWork — APPROVED** [review]

**Comment by asaxsma** [comment]
@mlemm99 , @driscoll85, @asaxsma - Reminder to review pull request.  Thanks!

**Review by driscoll85 — APPROVED** [review]

**Review by mlemm99 — APPROVED** [review]

**Pull request merged**